### ISSUE TITLE
Compiler crash when variable has same name as a function with overloads

### ISSUE THREAD

**Issue by DanielRosenwasser** [Microsoft/TypeScript#5160]
```TypeScript
var f = 10;

export function f();
export function f() {
}
```

```
Description:

getSemanticDiagnostics('C:\Users\drosen\AppData\Local\Temp\~vs212F.ts')

StackTrace:

Error: Debug Failure. False expression: 
   at assert (:1559:17)
   at reportImplementationExpectedError (:22811:29)
   at checkFunctionOrConstructorSymbol (:22902:17)
   at checkFunctionLikeDeclaration (:23399:21)
   at checkFunctionDeclaration (:23366:17)
   at checkSourceElement (:25187:21)
   at forEach (:866:17)
   at checkSourceFileWorker (:25380:17)
   at checkSourceFile (:25362:13)
   at getDiagnosticsWorker (:25422:17)
   at getDiagnostics (:25413:17)
   at Anonymous function (:35824:17)
   at runWithCancellationToken (:35800:17)
   at getSemanticDiagnosticsForFile (:35820:13)
   at getDiagnosticsHelper (:35775:17)
   at getSemanticDiagnostics (:35790:13)
   at getSemanticDiagnostics (:43916:13)
   at Anonymous function (:49157:17)
   at simpleForwardCall (:49016:9)
   at forwardJSONCall (:49032:13)
   at forwardJSONCall (:49078:13)
   at getSemanticDiagnostics (:49156:13)


---------------------------
Abort   Retry   Ignore   
---------------------------

**Comment by DanielRosenwasser** [comment]
@mhegazy this causes the compiler to crash. I think it should go in 1.7

**Comment by mhegazy** [comment]
let's get a fix ready first, then see what we can do.

**Comment by DanielRosenwasser** [comment]
I think Vlad fixed this in #5662 but I'd like to try it out and check in the tests.
